internal/cmdrender/executor.go

```diff
@@ -77,24 +84,39 @@ func (e *Executor) Execute(ctx context.Context) error {
 		return err
 	}
 
-	pkgWriter := &kio.LocalPackageReadWriter{PackagePath: string(root.pkg.UniquePath)}
-
 	// format resources before writing
 	_, err = filters.FormatFilter{UseSchema: true}.Filter(resources)
 	if err != nil {
 		return err
 	}
 
-	err = pkgWriter.Write(resources)
-	if err != nil {
-		return fmt.Errorf("failed to save resources: %w", err)
+	if e.Output == nil {
+		pkgWriter := &kio.LocalPackageReadWriter{PackagePath: string(root.pkg.UniquePath)}
+		err = pkgWriter.Write(resources)
+		if err != nil {
+			return fmt.Errorf("failed to save resources: %w", err)
+		}
+	} else {
+		// write to the output instead of the pkg directory if e.Output is specified
+		writer := &kio.ByteWriter{
+			Writer:                e.Output,
+			KeepReaderAnnotations: true,
+			WrappingAPIVersion:    kio.ResourceListAPIVersion,
+			WrappingKind:          kio.ResourceListKind,
+		}
+		err = writer.Write(resources)
+		if err != nil {
+			return fmt.Errorf("failed to write resources: %w", err)
+		}
 	}
 
 	if err = pruneResources(hctx); err != nil {
 		return err
 	}
 
-	pr.Printf("Successfully executed %d function(s) in %d package(s).\n", hctx.executedFunctionCnt, len(hctx.pkgs))
+	if !disableCLIOutput {
+		pr.Printf("Successfully executed %d function(s) in %d package(s).\n", hctx.executedFunctionCnt, len(hctx.pkgs))
+	}
```

Comment:
Actually this can also be moved in the `in-place` block.

Reply:
Yes. Updated.